quantumblacklabs/kedro-viz
# [KED-1883] Add automated testing for npm package import

## Description

Create a testing suite that automatically simulates publishing by exporting the package as a tar.gz file with npm pack, then imports it into a boilerplate react project and checks that it runs.
![image](https://user-images.githubusercontent.com/1155816/89541126-db094a80-d7f5-11ea-9f50-8cf28c690849.png)

## Development notes

- [x] Automatically `npm pack` then copy tarball to test project directory and install it
- [x] Create a local webpack dev server for manually QAing the test react project with kedro-viz imported
- [x] Write some automated tests for CI

## QA notes

This project should be very useful for QAing KedroViz in future. Run `npm run lib-test` to pack, copy, install and serve the test server. You can test updating the dataset by toggling the radio buttons.

You can run tests on the test react app by changing directory to `/tools/lib-test/react-app` and running `npm test`.

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
